incubator/hnc/internal/validators/object.go

```diff
@@ -139,6 +143,30 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	return o.handleInherited(op, newSource, oldSource, inst, oldInst)
 }
 
+func validateSelectorAnnot(inst *unstructured.Unstructured) string {
+	annots := inst.GetAnnotations()
+	for key, _ := range annots {
+		// for example: segs = ["propagate.hnc.x-k8s.io", "select"]
+		segs := strings.SplitN(key, "/", 2)
+		// for example: prefix = ["propagate", "hnc.x-k8s.io"]
+		prefix := strings.SplitN(segs[0], ".", 2)
+		if prefix[1] != api.MetaGroup {
```

Comment:
Seems it would get `panic: runtime error: index out of range [1] with length 1` if the annotation `key` is a plain word without a `.` in it.